src/utils/settingUtils.ts

```diff
@@ -33,16 +51,49 @@ export namespace settingUtils {
     }
 
     /**
-     * Uses ext.prefix 'azureResourceGroups' unless otherwise specified
+     * Directly retrieves one of the user's `Workspace` or `WorkspaceFolder` settings.
+     * @param key The key of the setting to retrieve
+     * @param fsPath The optional path of the workspace configuration settings
+     * @param targetSetting The optional workspace setting to target. Uses the `Workspace` configuration target unless otherwise specified
+     * @param prefix The optional extension prefix. Uses ext.prefix `containerApps` unless otherwise specified
      */
-    export function getWorkspaceSetting<T>(key: string, fsPath?: string, prefix: string = ext.prefix): T | undefined {
+    export function getWorkspaceSettingDirectly<T>(
+        key: string,
+        fsPath?: string,
+        targetSetting: ConfigurationTarget.Workspace | ConfigurationTarget.WorkspaceFolder = ConfigurationTarget.Workspace,
+        prefix: string = ext.prefix
+    ): T | undefined {
         const projectConfiguration: WorkspaceConfiguration = workspace.getConfiguration(prefix, fsPath ? Uri.file(fsPath) : undefined);
+        const result = projectConfiguration.inspect<T>(key);
+        return result && targetSetting === ConfigurationTarget.Workspace ? result?.workspaceValue : result?.workspaceFolderValue;
+    }
+
+    /**
+     * Iteratively retrieves one of the user's workspace settings - sequentially checking for a defined value starting from the `WorkspaceFolder` up to the provided target configuration limit.
+     * @param key The key of the setting to retrieve
+     * @param fsPath The optional path of the workspace configuration settings
+     * @param targetLimit The optional target configuration limit (inclusive). Uses the `Workspace` configuration target unless otherwise specified
+     * @param prefix The optional extension prefix. Uses ext.prefix `containerApps` unless otherwise specified
+     */
+    export function getWorkspaceSettingIteratively<T>(
```

Comment:
I thought it would be nice to be able to differentiate the behavior of picking a configuration target value `directly` vs. using VS Code's `get` command to `iteratively` search for a settings value from most to least specific.